sigmastate/src/main/scala/sigmastate/serialization/DataSerializer.scala

```diff
@@ -114,10 +114,7 @@ object DataSerializer {
         SigmaDsl.avlTree(AvlTreeData.serializer.parse(r))
       case tColl: SCollectionType[a] =>
         val len = r.getUShort()
-        if (tColl.elemType == SByte)
-          Colls.fromArray(r.getBytes(len))
-        else
-          deserializeColl(len, tColl.elemType, r)
+        deserializeColl(len, tColl.elemType, r)
```

Comment:
Please describe why this change is not breaking (better, do a test).

Reply:
This is covered by test, so not tests are necessary. 
Actually, the coverage tool shows that `case SByte =>` in `deserializeColl` is never executed, because it is covered by `if (tColl.elemType == SByte)`.

So if you can compare the following fragments to see this is not breaking change.
```
if (tColl.elemType == SByte)
          Colls.fromArray(r.getBytes(len))
```
 and 
```
case SByte =>
        Colls.fromArray(r.getBytes(len)).asInstanceOf[Coll[T#WrappedType]]
```